api/utils/retryutils/jitter_test.go

```diff
@@ -112,3 +123,34 @@ type mockInt63n func(n int64) int64
 func (m mockInt63n) Int63n(n int64) int64 {
 	return m(n)
 }
+
+// BenchmarkJitter is an attempt to check the effect of concurrency on the performance
+// of a global jitter instance. I'm a bit skeptical of how "true to life" this benchmark
+// really is, but the results would seem to indicate that >100k concurrent jitters would
+// still all complete in <1s, which is very good for our purposes.
+func BenchmarkSingleJitter(b *testing.B) {
+	benchmarkSharedJitter(b, NewHalfJitter())
+}
+
+func BenchmarkShardedJitter(b *testing.B) {
+	benchmarkSharedJitter(b, NewShardedHalfJitter())
+}
+
+func benchmarkSharedJitter(b *testing.B, jitter Jitter) {
+	benchmarkJitter(b, func() Jitter { return jitter })
+}
+
+func benchmarkJitter(b *testing.B, mkjitter func() Jitter) {
+	procs := runtime.GOMAXPROCS(0)
+	for n := procs; n < 200_000; n = n * 2 {
```

Comment:
Shouldn't you be using `b.N` here so the benchmark terminates at the right time?

Reply:
We're using `b.Run` below, those are the actual benchmarks.  This loop is building benchmarks with various amounts of concurrency, so that we can benchmark the effect of contention at various levels.

Reply:
Right, I guess I'm just concerned that you're not using `b.N` at all and my understanding is that the runtime will dynamically alter N so that it runs enough iterations until it is satisfied with the stability of the benchmark. Without using N and just running a fixed number of iterations, you may get misleading results due to variance in the code under test.

Reply:
Since we use `RunParallel` within our sub-benchmarks, we use `pb.Next()` instead of `b.N`, because the benchmarker distributes N across multiple parallel invocations of the supplied closure.  It's the same thing, just a different usage pattern.

Reply:
Today I learned!